src/main/java/org/galatea/starter/entrypoint/PriceRestController.java

```diff
@@ -19,27 +30,64 @@
  */
 @RequiredArgsConstructor
 @Slf4j
-@Log(enterLevel = Level.INFO, exitLevel = Level.INFO)
+@Log (enterLevel = Level.INFO, exitLevel = Level.INFO)
 @RestController
-
-
 public class PriceRestController extends BaseRestController {
 
 
   @NonNull
   PriceService priceService;
+  long totalTime;
+  String start;
 
   /**
    * Send the received text to the PriceService to be processed and send the result out
+   *
+   * @GetMapping to link http GET request to this method
+   * @RequestParam to take a parameter from the url
+   * @return
    */
-  // @GetMapping to link http GET request to this method
-  // @RequestParam to take a parameter from the url
+
+
   @GetMapping(value = "${webservice.halpath}", produces = {MediaType.APPLICATION_JSON_VALUE})
-  public String priceEndpoint(
-      @RequestParam(value = "stock") String stock,
-      @RequestParam(value = "daysToLookBack", defaultValue= "1", required = false) int daysToLookBack,
-      @RequestParam (value = "requestId", required = false) String requestId) {
+  public FullResponse priceEndpoint(
+
+  @RequestParam(value = "stock") String stock,
+      @RequestParam(value = "days", defaultValue= "1", required = false) String daysToLookBack,
+      @RequestParam (value = "requestId", required = false) String requestId)
+
+      throws IOException{
+
+    long processStartTime = System.currentTimeMillis();
+    start = DateFormat.getInstance().format(processStartTime);
     processRequestId (requestId);
-    return priceService.processStock(stock, daysToLookBack);
+    priceService.getPricesByStock(stock, daysToLookBack);
+    long processEndTime = System.currentTimeMillis();
+    totalTime = processEndTime - processStartTime;
+    return buildFullResponse(stock, daysToLookBack, priceService.responseTime );
+    }
+
+  public FullResponse buildFullResponse(String stock, String days, Long reponseTime)
+      throws UnknownHostException{
+
+    StockMetadata stockMetadata;
+    ArrayList<StockMetadata> metadata = new ArrayList<>();
+
+    //Construct Meta Data object
+    StockMetadataBuilder builder = StockMetadata.builder();
+    builder.endpoint("price?stock=" + stock + "&days=" + days);
+    builder.host(InetAddress.getLocalHost().getHostName());
+    builder.processTime(totalTime + "(ms) ");
+    builder.responseTime(reponseTime + "(ms)");
+    builder.timeStamp(start);
+    stockMetadata = builder.build();
+
+    metadata.add(stockMetadata);
+
+
+    // Create complete response object and pretty print JSON
+    FullResponse fullResponse = new FullResponse(metadata, priceService.filteredPrices);
```

Comment:
pass in the prices as a parameter. start a chain where they get returned from the call  to getPricesByStock and then get fed into here.